napari/layers/base/base.py

```diff
@@ -637,61 +641,23 @@ def _is_moving(self, value):
             assert self._moving_coordinates is not None
         self._private_is_moving = value
 
-    @property
-    def _dims_displayed(self):
-        """To be removed displayed dimensions."""
-        # Ultimately we aim to remove all slicing information from the layer
-        # itself so that layers can be sliced in different ways for multiple
-        # canvas. See https://github.com/napari/napari/pull/1919#issuecomment-738585093
-        # for additional discussion.
-        return self._dims_order[-self._ndisplay :]
-
-    @property
-    def _dims_not_displayed(self):
-        """To be removed not displayed dimensions."""
-        # Ultimately we aim to remove all slicing information from the layer
-        # itself so that layers can be sliced in different ways for multiple
-        # canvas. See https://github.com/napari/napari/pull/1919#issuecomment-738585093
-        # for additional discussion.
-        return self._dims_order[: -self._ndisplay]
-
-    @property
-    def _dims_displayed_order(self):
-        """To be removed order of displayed dimensions."""
-        # Ultimately we aim to remove all slicing information from the layer
-        # itself so that layers can be sliced in different ways for multiple
-        # canvas. See https://github.com/napari/napari/pull/1919#issuecomment-738585093
-        # for additional discussion.
-        displayed = self._dims_displayed
-        # equivalent to: order = np.argsort(displayed)
-        order = sorted(range(len(displayed)), key=lambda x: displayed[x])
-        return tuple(order)
-
     def _update_dims(self, event=None):
         """Update the dims model and clear the extent cache.
 
         This function needs to be called whenever data or transform information
         changes, and should be called before events get emitted.
         """
-        from ...components.dims import reorder_after_dim_reduction
-
         ndim = self._get_ndim()
 
         old_ndim = self._ndim
         if old_ndim > ndim:
             keep_axes = range(old_ndim - ndim, old_ndim)
             self._transforms = self._transforms.set_slice(keep_axes)
```

Comment:
I'm not very familiar with the transform system, but I noticed you left this stateful update outside of the `SliceInput`. Is that inevitable? Why do transforms themselves need to change when slicing changes?

Reply:
Great question. When we do slicing `_update_dims` is called, but neither of these conditions will be true, so the transforms won't actually be modified.

I have it on my bucketlist to clean up / reduce use of `_update_dims`, but not in this PR. I think it's only really needed when `data` is set, since the new `data` could actually have new dimensionality. We also call it in the transform property setters, which is to clear our the extent cache, but again dimensionality should not actually change.

Reply:
I see; Is it poossible to generalize transforms so that if you combine a 3D affine with a 4D affine, everything works? This way we wouldn't have to wirry about this at the layer level...

Reply:
> I see; Is it poossible to generalize transforms so that if you combine a 3D affine with a 4D affine, everything works? This way we wouldn't have to wirry about this at the layer level...

Maybe - depends what you mean by everything works. Personally, I want to be able to explicitly assign dimensions to the layer's transform. If someone passes a transform that does not explicitly account for some dimension, we can have some logic to handle that, but in some cases should just error. In which case, I prefer to raise early and handle things at the layer level. Otherwise, we'll error when trying to make correspondences between layers which seems like it would be confusing.

Remember, this section isn't really about slicing. It's about `data` changing.